### PR TITLE
Autobuild: Use macOS 11 and Xcode 12.5.1

### PR DESCRIPTION
**Pull request by hoffie**
**Short description of changes**

- Autobuild: Build on macOS 11
  We previously couldn't run on macOS 11 as dmgbuild did not support it.
  Now that we are using create-dmg, this blocker is gone.

- Autobuild: Bump Mac Xcode version to 12.5.1
  Previously, we couldn't go past 12.4.x as newer versions required macOS 11. As we will be on macOS 11 now, we can use the latest Xcode version which is supposrted by Qt5.

CHANGELOG: Autobuild: Updated build environments to macOS 11 and Xcode 12.5.1

**Context: Fixes an issue?**

No, just trying hard to use the most-supported environments.

**Does this change need documentation? What needs to be documented and how?**

No.

**Status of this Pull Request**

Ready.

**What is missing until this pull request can be merged?**

- Mac test run (@softins)?
- Approvals

## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [ ] I tested my code and it does what I want
- [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I've filled all the content above

### CODE REVIEW

**Review by ann0see — COMMENTED** [review]
I'd assume we can't really test this. What should be ensured is that the minimal supported macOS version isn't changed.

**Comment by hoffie** [comment]
After some more reading:
- The switch from Xcode 12.1.1 to 12.5.1 does update the macOS SDK [from `macOS 10.15.6` to `macOS 11.3`](https://developer.apple.com/support/xcode/). Note though that this is not equivalent to dropping support for earlier version than 11.3 according to these docs: https://firefox-source-docs.mozilla.org/widget/cocoa/sdks.html
- At the same time, bugs are to be expected: https://forum.juce.com/t/targeting-older-macos-sdks-with-xcode-12/45217

So my tendency right now would be: Point the users who experienced macOS 11 issues to the artifacts of this PR. If this magically fixed those issues, we should certainly take the risk of having annoyances with older macOS releases in favor of fixing actual issues with newer releases. If there are no visible improvements, I'd close this PR (and would open another for updating to macOS 11 only).

@emlynmac @softins Any advice on what's recommended in general?

**Comment by softins** [comment]
> @emlynmac @softins Any advice on what's recommended in general?

I'm afraid I'm not an expert on MacOS development and xcode/SDK versions. I have to google when I want to know about them!

File: .github/workflows/autobuild.yml [hunk 1]
```diff
@@ -111,22 +111,24 @@ jobs:
...
-             building_on_os:        macos-10.15
+             building_on_os:        macos-11
              cmd1_prebuild:         "./autobuild/mac/artifacts/autobuild_mac_1_prepare.sh 5.15.2"
              cmd2_build:            "./autobuild/mac/artifacts/autobuild_mac_2_build.sh sign_if_possible"
              cmd3_postbuild:        "./autobuild/mac/artifacts/autobuild_mac_3_copy_files.sh"
              uses_codeql:           false
-             xcode_version:         12.1.1
+             # Qt5's maximum supported Xcode version is 12.x (SDK 11). Therefore, use the latest
+             # 12.x release here.
+             # https://xcodereleases.com/ https://doc.qt.io/qt-5/macos.html
```
Comment (softins) [inline .github/workflows/autobuild.yml:130]: I see there is still a build warning:

```
Project WARNING: Qt has only been tested with version 10.15 of the platform SDK, you're using 11.3.
Project WARNING: This is an unsupported configuration. You may experience build issues, and by using
Project WARNING: the 11.3 SDK you are opting in to new features that Qt has not been prepared for.
Project WARNING: Please downgrade the SDK you use to build your app to version 10.15, or configure
Project WARNING: with CONFIG+=sdk_no_version_check when running qmake to silence this warning.
```

That is at odds with what is stated at https://doc.qt.io/qt-5/macos.html

It makes me wonder whether SDK11 compatibility was introduced in one of the proprietary Qt 5.15.x where x > 2?

If thorough testing shows the built artefact does indeed run properly, maybe we should add the specified `CONFIG+=sdk_no_version_check` when building this version?

Comment (hoffie) [reply .github/workflows/autobuild.yml:130]: > I see there is still a build warning:

Thanks, I've missed this as I was relying on the documented behavior.

> It makes me wonder whether SDK11 compatibility was introduced in one of the proprietary Qt 5.15.x where x > 2?

Possible, yes.

> If thorough testing shows the built artefact does indeed run properly, maybe we should add the specified CONFIG+=sdk_no_version_check when building this version?

Yes, if we continue this path we should definitely add this and explain why we did it. However, as long as there are no positive reports about the Xcode update fixing things, this warning reinforced my tendency not to perform this change right now.

Therefore, I'm going to close this PR. If we have reports that it helps, I'll reopen and add the `sdk_no_version_check`.
I'll open a seperate PR for `macos-11` as we should do that anyway.

**Pull request closed without merging**